### PR TITLE
AT-SSO-UI 31 Refactor at-sce-ui to at-sso-ui, USERS

### PR DESCRIPTION
**Pull request by Emiliano-Plascencia**
We refactor the code from at-sce-ui to at-sso-ui
We migrate the code related with the users (views, functions, CRUD operations, routes, services, unit test, etc)

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -6,21 +6,53 @@ const morgan = require("morgan");
...
+      typeProject: function (a) {
+        if ( a == undefined || a == 0 ) return ""
+        else if ( a == 1 ) return "Custome Web"
+        else if ( a == 2 ) return "e-commerce"
+        else if ( a == 3 ) return "Android app"
+        else if ( a == 4 ) return "IOS app";
+      },
+      formatCurrency: function (a) {
+        return a.toString().replace(/(\d)(?=(\d\d\d)+(?!\d))/g, "$1,");
+      },
```
Comment (irvingesp2190) [inline src/server.js]: I think this code is not necessary in SSO

**Review by moselGlez — APPROVED** [review]

**Review by FlavioAlejandro — APPROVED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (at-internship) [inline .vscode/settings.json]: This file is not necessary

File: src/config/passport.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+/**
+ * AT SCE UI - AT Auth Passport.
```
Comment (at-internship) [inline src/config/passport.js]: A SSO UI

File: src/controllers/admin.controller.js [hunk 4]
```diff
@@ -1,8 +1,29 @@
+/**
+ * AT SCE UI - AT Admin Controller.
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: AT SSO UI

File: src/controllers/at-sso.controller.js [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
 };
 
-ssoCtrl.signin = async(req, res) => {
-    console.log("--> ssoCtrl.signin");
-
-    // Redirect
-    req.flash("success_msg", "User signin Successfully");
-    res.redirect("/home");
-};
+ssoCtrl.signin = passport.authenticate("local", {
```
Comment (at-internship) [inline src/controllers/at-sso.controller.js]: // AT-SSO - Signin

File: src/routes/at-sso.routes.js [hunk 6]
```diff
@@ -4,6 +4,7 @@ const path = require('path');
 
 // Admin Controller
 const { renderSigninForm, signin, signout, home, team } = require("../controllers/at-sso.controller");
+const { isAuthenticated } = require("../helpers/auth.helper");
 
 // Helpers
 // const { isAdmin } = require("../helpers/auth");
```
Comment (at-internship) [inline src/routes/at-sso.routes.js]: Delete

File: src/server.js [hunk 7]
```diff
@@ -6,21 +6,53 @@ const morgan = require("morgan");
...
+      selected: function (a, b) {
+        if (a == undefined) return "";
+        return a == b ? "selected" : "";
+      },
+      statusHelper: function (a) {
+        if (a == undefined) return "";
+        return a == 1 ? "Active" : "Inactive";
+      },
+      typeProject: function (a) {
```
Comment (at-internship) [inline src/server.js]: These helpers are not necessary: typeProject & formatCurrency

File: src/services/at-sso-api.service.js [hunk 8]
```diff
@@ -12,20 +12,44 @@
 const axios = require("axios");
 const AT_SSO_SERVICE = {};
 
+// MICROSERVICE - HEROKU - SCE
+const AT_SCE_SERVICE_URI = process.env.AT_SCE_SERVICE_URI;
+console.debug(`at-sce-api.service - AT_SCE_SERVICE_URI: ${AT_SCE_SERVICE_URI}`);
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: `console.debug(`at-sso-api.service - AT_SCE_SERVICE_URI: ${AT_SCE_SERVICE_URI}`);`

File: src/services/at-sso-api.service.js [hunk 9]
```diff
@@ -12,20 +12,44 @@
...
+
+// MICROSERVICE - HEROKU - SS0
+const AT_SSO_SERVICE_URI = process.env.AT_SSO_SERVICE_URI;
+console.debug(`at-sce-api.service - AT_SSO_SERVICE_URI: ${AT_SSO_SERVICE_URI}`);
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: `console.debug(`at-sso-api.service - AT_SSO_SERVICE_URI: ${AT_SSO_SERVICE_URI}`);`

File: src/services/at-sso-api.service.js [hunk 10]
```diff
@@ -12,20 +12,44 @@
...
+
+// AT_SSO_SERVICE_URI_ENABLED FLAG
+const AT_SSO_SERVICE_URI_ENABLED = process.env.AT_SSO_SERVICE_URI_ENABLED;
+const AT_SERVICE_URI = (AT_SSO_SERVICE_URI_ENABLED == 'true') ? AT_SSO_SERVICE_URI : AT_SCE_SERVICE_URI;
+console.log(`at-sce-api.service - AT_SERVICE_URI: ${AT_SERVICE_URI}`);
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: console.log(`at-sso-api.service - AT_SERVICE_URI: ${AT_SERVICE_URI}`);

File: src/services/at-sso-api.service.js [hunk 11]
```diff
@@ -12,20 +12,44 @@
...
+
 // LOCAL
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: Remove code comments

File: src/services/at-sso-api.service.js [hunk 12]
```diff
@@ -12,20 +12,44 @@
...
 //require("dotenv").config();
 //const AT_SSO_SERVICE_URI = process.env.AT_SSO_SERVICE_URI || `https://at-sso-api.herokuapp.com/api`;
 
 // PROD
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: Remove code comments

File: src/services/at-sso-api.service.js [hunk 13]
```diff
@@ -34,15 +58,60 @@ AT_SSO_SERVICE.getAllUsers = () => {
...
+AT_SSO_SERVICE.deleteUser = (id) => {
+  return axios({
+    method: "DELETE",
+    url: `${AT_SERVICE_URI}/v1/users/${id}`,
+    headers: {
+      "content-type": "application/json",
+    },
+  });
+};
+//Add user
```
Comment (at-internship) [inline src/services/at-sso-api.service.js]: addUser - is this operation still necessary?

File: src/test/controllers/admin.controller.test.js [hunk 14]
```diff
@@ -18,32 +18,38 @@ const adminController = require("../../controllers/admin.controller");
...
         getAllUsersStub.restore();
         addUserStub.restore();
+        getUserByIdStub.restore();
+        updateUserStub.restore();
+        deleteUserStub.restore();
     });
 
- // AT-SSO - Admin - Index
-    it("Should render admin Index", function(done) {
+    // AT-SCE - Admin - Index
```
Comment (at-internship) [inline src/test/controllers/admin.controller.test.js]: `// AT-SSO - Admin - Index`

why? `// AT-SCE - Admin - Index`

File: src/views/admin/index.hbs [hunk 15]
```diff
@@ -1,32 +1,32 @@
...
-        </table>
-    </div>
-</section>
\ No newline at end of file
+     <section class="card">
+          <div class="card-body">
+               <table style="width: 100%">
+                    <tr>
+                         <td style="text-align: left;">
+                              <h4>AT-SCE</h4>
```
Comment (at-internship) [inline src/views/admin/index.hbs]: ``<h4>AT-SSO</h4>

File: src/views/layouts/main.hbs [hunk 16]
```diff
@@ -14,8 +14,8 @@
     <!--<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.20/css/jquery.dataTables.min.css"/>
     <script type="text/javascript" src="https://cdn.datatables.net/1.10.20/js/jquery.dataTables.min.js"></script>-->
 
-    <!-- BOOTSTRAP - BOOTSWATCH THEME - SPACELAB -->
-    <link rel="stylesheet" href="https://bootswatch.com/4/spacelab/bootstrap.min.css">
+    <!-- BOOTSTRAP - BOOTSWATCH THEME - LITERA -->
+    <link rel="stylesheet" href="https://bootswatch.com/4/litera/bootstrap.min.css">
```
Comment (at-internship) [inline src/views/layouts/main.hbs]: why did u change this?

File: src/views/partials/navigation.hbs [hunk 17]
```diff
@@ -16,18 +16,7 @@
...
-                        <span class="d-lg-none">AT University</span>
-                    </a>
-                </li>
-                <li class="nav-item">
-                    <a class="nav-link" href="https://at-resources-ui.herokuapp.com/dashboard">
-                        <span class="d-none d-lg-block">AT Resources</span>
-                        <span class="d-lg-none">AT Resources</span>
-                    </a>
-                </li>
+
```
Comment (at-internship) [inline src/views/partials/navigation.hbs]: why did u change this?

**Review by at-internship — CHANGES_REQUESTED** [review]

File: src/views/layouts/main.hbs (hunk 16 above)
Comment (moselGlez) [reply src/views/layouts/main.hbs]: We changed this becuse Litera is the current style in SCE

**Comment by at-internship** [comment]
Additionally, files with a txt extension are not necessary

**Review by jaimeIxba — APPROVED** [review]

**Review by irvingesp2190 — APPROVED** [review]

**Review by at-internship — CHANGES_REQUESTED** [review]
It is necessary to integrate the web token, encrypt and unit test module of the at-sce-ui project

**Review by FlavioAlejandro — APPROVED** [review]

**Review by DianaArceoSuarez — APPROVED** [review]

**Review by at-internship — APPROVED** [review]

**Pull request merged**